src/Rule/Nested.php

```diff
@@ -132,18 +135,13 @@ private function prepareRules(iterable|object|string|null $source): ?iterable
 
         if ($source instanceof RulesProviderInterface) {
             $rules = $source->getRules();
-            return $this->normalizeRules ? $this->normalizeRules($rules) : $rules;
```

Comment:
Why we don't can make return here?

Reply:
To perform check and normalization for all cases and prevent duplication of corresponding code. Or can we guarantee that in other cases it's not needed? Maybe normalization can be skipped in certain scenarios, but what about checking? I suggest to leave it like that.